hibernate-core/src/main/java/org/hibernate/type/descriptor/java/ByteArrayTypeDescriptor.java

```diff
@@ -67,7 +67,7 @@ public String toString(Byte[] bytes) {
 		Byte[] bytes = new Byte[string.length() / 2];
 		for ( int i = 0; i < bytes.length; i++ ) {
 			final String hexStr = string.substring( i * 2, (i + 1) * 2 );
-			bytes[i] = (byte) ( Integer.parseInt( hexStr, 16 ) + Byte.MIN_VALUE );
+			bytes[i] = (byte) Integer.parseInt( hexStr, 16 );
```

Comment:
This old code wasn't consistent with how JDBC drivers handle the conversion from unsigned to signed bytes.